sunpy/coordinates/tests/test_sun.py

```diff
@@ -501,7 +502,7 @@ def test_carrington_rotation_time_longitude_numpy(crot, longitude, crot_fraction
                              (2000.5, 180),
 ])
 def test_carrington_rotation_time_longitude_err(crot, longitude):
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match=re.escape("Carrington longitude(s) must be > 0 degrees and <= 360 degrees.") + "|" + re.escape("Carrington rotation number(s) must be integral if `longitude` is provided.")):
```

Comment:
This is a general comment, you do not need to match the entire message, you can match a subset instead and that will make this much easier and shorter for you.

Reply:
Respected @nabobalis 
Thank you for your input! I highly value your feedback. I will take care from next time. 

Regards

Reply:
I don't think this should be merged with the full length match statements as they are now. 

Others may or may not agree but it's a lot of changes when we just need to match the relevant part of the message and involves specific regex escaping.

Reply:
Respected @nabobalis 
I agree. I tried to but in some cases there were problems. Some were straightforward while others required some tinkering. There were about 70 statements in total in the code that needed to be fixed. I have tested it on my machine and all tests have passed. I will try to optimize this in the future as needed. However, as an initial fix, this seems to be fine. However, I agree there should be a thorough code-review of this before merging this. 

Thank you sir for your time and guidance. I highly appreciate it. It motivates me to keep on contributing and learning about open source. 

Regards

Reply:
Just a few tips as you are new.

You do not need to add "Respected" when you @ someone and nor do you need to add "sir".

Reply:
@nabobalis I understand. Thank you for clarifying.